### PR TITLE
Make el consistent with bcadmin

### PR DESCRIPTION
**Pull request by kc1212**
- use the bcadmin library for operations in el
- fix integration test (flakiness on jenkins was observed locally too, it should be fixed on this PR)
- the OpenID authentication are untested and left as before, they will be fixed in a future PR

Discussion point:
I copy pasted the key function from bcadmin to el because the keys for logging events need to stay in the el config directory. Alternatively, we tell el the bcadmin config directory by adding a flag (e.g., `--bcconfig`) and remove the `key` function from el. The first approach has nicer UI, but the second approach removes repeated functionality.

### CODE REVIEW

**Comment by nkcr** [comment]
> we tell el the bcadmin config directory

Isn't it the purpose of the global `config` param / `BC_CONFIG` env var ?

**Comment by nkcr** [comment]
> we tell el the bcadmin config directory by adding a flag (e.g., `--bcconfig`) and remove the `key` function from el

Can't we use bcadmin and specify the el directory with the `--config` param ?

**Comment by kc1212** [comment]
> > we tell el the bcadmin config directory by adding a flag (e.g., `--bcconfig`) and remove the `key` function from el
> 
> Can't we use bcadmin and specify the el directory with the `--config` param ?

You could, but the user need to be cautious because he/she shouldn't use `--config` for non-eventlog keys. Actually, now that I think about it, when using `el` in production, the user is probably on a different machine than the one running `bcadmin`. The `el` user might not even have the `bcadmin` tool. So it needs to create the key using `el key`, send the public part to the administrator and then the administrator will send back the new DARC. This way, the `el` user doesn't need to switch between `el` and `bcadmin`. In my opinion we should keep the `key` subcommand in `el` for better user experience.

File: eventlog/el/test.sh [hunk 1]
```diff
@@ -18,52 +18,42 @@ main(){
...
-	eval $SED
-	[ -z "$PRIVATE_KEY" ] && exit 1
-	ID=`awk '/^Identity: / { print $2}' < $RUNOUT`
-	[ -z "$ID" ] && exit 1
-
+testEventLog(){
+	##### setup phase
 	runCoBG 1 2 3
 	# block interval of 2 seconds to this particular test to fail because of small intervals
 	# (Note: might go back to 0.5 after #1813)
```
Comment (jeffallen) [inline eventlog/el/test.sh]: #1813 is fixed, remove comment? reduce interval?

File: eventlog/README.md [hunk 2]
```diff
@@ -39,67 +38,63 @@ An example transcript of correctly setting up and using an Eventlog is:
 ```
 # make the ByzCoin instance
```
Comment (jeffallen) [inline eventlog/README.md]:
```suggestion
# Make the ByzCoin instance
```

File: eventlog/README.md [hunk 3]
```diff
@@ -112,9 +107,5 @@ long now = System.currentTimeMillis() * 1000 * 1000;
 SearchResponse resp = el.search("", now - 1000, now + 1000);
 ```
 
-Please refer to the javadocs for more information. The javadocs are not hosted
```
Comment (jeffallen) [inline eventlog/README.md]: Want to give a link to javadoc.io?

Comment (kc1212) [reply eventlog/README.md]: the java doc is given above https://github.com/dedis/cothority/pull/1911/files#diff-3ff608f4768b8ab0247453c05725047fR76

**Pull request merged**